File: 5. Longest Palindromic Substring/Solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = ''
        for i in range(2 * len(s) - 1):
            left = i // 2
            right = left + i % 2

            while left >= 0 and right < len(s) and s[left] == s[right]:
                temp = s[left:right+1]
                if len(temp) > len(res):
                    res = temp
                left -= 1
                right += 1
        return res

File: 5. Longest Palindromic Substring/Solution.java
class Solution {
    public String longestPalindrome(String s) {
        int len = s.length();
        char[] charArr = s.toCharArray();
        if(s.length() < 2){
            return s;
        }
        boolean[][] dp = new boolean[len][len];
        for(int i = 0; i< len; i++){
            dp[i][i] = true;
        }
        int left = 0;
        int max = 1;

        for(int j = 1; j < len; j++){
            for(int i = 0; i < j; i++){
                if(charArr[i] != charArr[j]){
                    dp[i][j] = false; 
                }else{
                    if((j-1) - (i+1) + 1 < 2){
                        dp[i][j] = true;
                    }else{
                        dp[i][j] = dp[i+1][j-1];
                    }
                }
                if(dp[i][j] == true && j-i+1 > max){
                    max = j-i+1;
                    left = i;
                }
            }
        }
    return s.substring(left, left+max);
    }
}
